server/src/main/java/org/elasticsearch/action/admin/cluster/stats/MappingStats.java

```diff
@@ -48,85 +48,90 @@ public static MappingStats of(Metadata metadata, Runnable ensureNotCancelled) {
-                    } else if (scriptObject instanceof Map<?, ?> script) {
+                    if (scriptObject instanceof Map<?, ?> script) {
                         Object sourceObject = script.get("source");
-                        updateScriptParams(sourceObject, stats.fieldScriptStats);
+                        stats.scriptCount += count;
+                        updateScriptParams(sourceObject, stats.fieldScriptStats, count);
                         Object langObject = script.get("lang");
                         if (langObject != null) {
                             stats.scriptLangs.add(langObject.toString());
                         }
                     }
-                });
-            }
+                }
+            });
+
+            MappingVisitor.visitRuntimeMapping(map, (field, fieldMapping) -> {
+                Object typeObject = fieldMapping.get("type");
+                if (typeObject == null) {
+                    return;
+                }
+                String type = typeObject.toString();
+                RuntimeFieldStats stats = runtimeFieldTypes.computeIfAbsent(type, RuntimeFieldStats::new);
+                stats.count += count;
+                if (indexRuntimeFieldTypes.add(type)) {
+                    stats.indexCount += count;
+                }
+                if (concreteFieldNames.contains(field)) {
+                    stats.shadowedCount += count;
+                }
+                Object scriptObject = fieldMapping.get("script");
+                if (scriptObject == null) {
+                    stats.scriptLessCount += count;
+                } else if (scriptObject instanceof Map<?, ?> script) {
+                    Object sourceObject = script.get("source");
+                    updateScriptParams(sourceObject, stats.fieldScriptStats, count);
+                    Object langObject = script.get("lang");
+                    if (langObject != null) {
+                        stats.scriptLangs.add(langObject.toString());
+                    }
+                }
+            });
         }
         return new MappingStats(fieldTypes.values(), runtimeFieldTypes.values());
     }
 
-    private static void updateScriptParams(Object scriptSourceObject, FieldScriptStats scriptStats) {
+    private static void updateScriptParams(Object scriptSourceObject, FieldScriptStats scriptStats, int multiplier) {
         if (scriptSourceObject != null) {
             String scriptSource = scriptSourceObject.toString();
             int chars = scriptSource.length();
             long lines = scriptSource.lines().count();
             int docUsages = countOccurrences(scriptSource, DOC_PATTERN);
             int sourceUsages = countOccurrences(scriptSource, SOURCE_PATTERN);
-            scriptStats.update(chars, lines, sourceUsages, docUsages);
+            for (int i = 0; i < multiplier; i++) {
+                scriptStats.update(chars, lines, sourceUsages, docUsages);
```

Comment:
I think we can add a `multiplier` param to script states too to avoid the loop?

Reply:
++ that's cuter :)